app/src/main/java/com/xmartlabs/scasas/geoquiz/ui/QuizActivity.java

```diff
@@ -70,45 +54,162 @@ public boolean onOptionsItemSelected(MenuItem item) {
       @Override
       public void onClick(View v) {
-        Toast.makeText(QuizActivity.this, checkAnswer(trueButon.getText().toString()), Toast.LENGTH_SHORT).show();
+        if (!isLastQuestion()) {
+          currentIndex++;
+          questionTextView.setText(questionBank.get(currentIndex).getTextResId());
+          if (isLastQuestion()) {
+            nextButton.setEnabled(false);
+          }
+          if (!prevButton.isEnabled() && !isFirstQuestion()) {
+            prevButton.setEnabled(true);
+          }
+        } else {
+          Toast.makeText(QuizActivity.this, R.string.no_more, Toast.LENGTH_SHORT).show();
+        }
       }
     });
-    falseButton = (Button) findViewById(R.id.false_button);
-    falseButton.setOnClickListener(new View.OnClickListener() {
+  }
+
+  private void setupPrevButton() {
+    prevButton = (Button) findViewById(R.id.prev_button);
+    prevButton.setEnabled(false);
+    prevButton.setOnClickListener(new View.OnClickListener() {
       @Override
       public void onClick(View v) {
-        Toast.makeText(QuizActivity.this, checkAnswer(falseButton.getText().toString()), Toast.LENGTH_SHORT).show();
+        if (!isFirstQuestion()) {
+          if (!prevButton.isEnabled()) {
+            prevButton.setEnabled(true);
+          }
+          currentIndex--;
+          questionTextView.setText(questionBank.get(currentIndex).getTextResId());
+          if (!nextButton.isEnabled() && !isLastQuestion()) {
+            nextButton.setEnabled(true);
+          }
+          if (isFirstQuestion()) {
+            prevButton.setEnabled(false);
+          }
+        }
       }
     });
+  }
+
+  private void setupNextButton() {
     nextButton = (Button) findViewById(R.id.next_button);
     nextButton.setOnClickListener(new View.OnClickListener() {
       @Override
       public void onClick(View v) {
-        if (currentIndex == questionBank.length - 1) {
-          Toast.makeText(QuizActivity.this, "No more questions for today :)", Toast.LENGTH_SHORT).show();
-        } else {
+        if (!isLastQuestion()) {
+          if (!nextButton.isEnabled()) {
+            nextButton.setEnabled(true);
+          }
           currentIndex++;
-          questionTextView.setText(questionBank[currentIndex].getTextResId());
+          questionTextView.setText(questionBank.get(currentIndex).getTextResId());
+          if (!prevButton.isEnabled() && !isFirstQuestion()) {
+            prevButton.setEnabled(true);
+          }
+          if (isLastQuestion()) {
+            nextButton.setEnabled(false);
+          }
         }
       }
     });
   }
 
+  private void setupFalseButton() {
+    falseButton = (Button) findViewById(R.id.false_button);
+    falseButton.setOnClickListener(new View.OnClickListener() {
+      @Override
+      public void onClick(View v) {
+        Toast.makeText(QuizActivity.this, obtainAnswerValue(falseButton.getText().toString()), Toast.LENGTH_SHORT).show();
```

Comment:
what do you think of putting this sentence (and all the similar sentences) in a function? something like `private void showToast(@NonNull String message)`